js/interface.js

```diff
@@ -686,36 +698,45 @@ $('input[type="file"]').on('change', function() {
 $('.redirectToSettings, [data-change-settings]').on('click', function(event) {
   event.preventDefault();
 
-  Fliplet.Studio.emit('close-overlay', {
-    name: 'publish-google'
+  saveProgressOnClose().then(function () {
+    Fliplet.Studio.emit('close-overlay', {
+      name: 'publish-google'
+    });
+  
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  }).catch(function (error) {
+    Fliplet.Modal.alert({
+      message: Fliplet.parseError(error)
+    })
   });
 
-  Fliplet.Studio.emit('overlay', {
-    name: 'app-settings',
-    options: {
-      size: 'large',
-      title: 'App Settings',
-      section: 'appSettingsGeneral',
-      appId: Fliplet.Env.get('appId')
-    }
-  });
 });
 
 $('[data-change-assets]').on('click', function(event) {
   event.preventDefault();
 
-  Fliplet.Studio.emit('close-overlay', {
-    name: 'publish-google'
-  });
-
-  Fliplet.Studio.emit('overlay', {
-    name: 'app-settings',
-    options: {
-      size: 'large',
-      title: 'App Settings',
-      section: 'launchAssets',
-      appId: Fliplet.Env.get('appId')
-    }
+  saveProgressOnClose().then(function () {
+    Fliplet.Studio.emit('close-overlay', {
+      name: 'publish-google'
+    });
+  
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'launchAssets',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
   });
```

Comment:
There should be a `.catch()` clause here too.

Reply:
@tonytlwu 
Sorry I didn't notice that.
Fixed.